src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,738 @@
+
+    private fun getPlaceableSide(pos: BlockPos): EnumFacing? {
+        for (side in EnumFacing.values()) {
+            val neighbour = pos.offset(side)
+            if (!mc.world.getBlockState(neighbour).block.canCollideCheck(mc.world.getBlockState(neighbour), false)) {
+                continue
+            }
+            val blockState = mc.world.getBlockState(neighbour)
+            if (!blockState.material.isReplaceable) {
+                return side
+            }
+        }
+        return null
+    }
+
+    private fun lookAtBlock(pos: BlockPos) {
+        val vec3d = Vec3d(pos).add(0.5, 0.0, 0.5)
+        val lookAt = RotationUtils.getRotationTo(vec3d, true)
+        mc.player.rotationYaw = lookAt.x.toFloat()
+        mc.player.rotationPitch = lookAt.y.toFloat()
+    }
+
+    fun getPlayerDirection(): Int {
+        val yaw = (mc.player.rotationYaw % 360 + 360) % 360
+        return if (yaw >= 158 && yaw < 203) { 0 } //NORTH
+        else if (yaw >= 203 && yaw < 258) { 1 } //NORTH-EAST
+        else if (yaw >= 258 && yaw < 293) { 2 } //EAST
+        else if (yaw >= 293 && yaw < 338) { 3 } //SOUTH-EAST
+        else if (yaw >= 338 || yaw < 23) { 4 } //SOUTH
+        else if (yaw >= 23 && yaw < 68) { 5 } //SOUTH-WEST
+        else if (yaw >= 68 && yaw < 113) { 6 } //WEST
+        else { 7 } //NORTH-WEST
+    }
+
+    private fun updateRenderer(renderer: ESPRenderer): ESPRenderer {
+        val side = GeometryMasks.Quad.ALL
+        for (bt in blockQueue) {
+            if (bt.getTaskState() != TaskState.DONE) { renderer.add(bt.getBlockPos(), bt.getTaskState().color, side) }
+        }
+        for (bt in doneQueue) {
+            if (bt.getBlock()::class != BlockAir::class) { renderer.add(bt.getBlockPos(), bt.getTaskState().color, side) }
+        }
+        return renderer
+    }
+
+    override fun onWorldRender(event: RenderEvent) {
+        if (mc.player == null) return
+        val renderer = ESPRenderer()
+        renderer.aFilled = if (filled.value) aFilled.value else 0
+        renderer.aOutline = if (outline.value) aOutline.value else 0
+        updateRenderer(renderer)
+        renderer.render(true)
+    }
+
+    private fun printDebug() {
+        MessageSendHelper.sendChatMessage("")
+        MessageSendHelper.sendChatMessage("")
+        MessageSendHelper.sendChatMessage("-------------------- QUEUE -------------------")
+        for (bt in blockQueue) {
+            MessageSendHelper.sendChatMessage(bt.getBlock().localizedName + "@(" + bt.getBlockPos().asString() + ") State: " + bt.getTaskState().toString() + " Prio: " + bt.getPriority())
+        }
+        MessageSendHelper.sendChatMessage("")
+        MessageSendHelper.sendChatMessage("-------------------- DONE --------------------")
+        for (bt in doneQueue) {
+            MessageSendHelper.sendChatMessage(bt.getBlock().localizedName + "@(" + bt.getBlockPos().asString() + ") State: " + bt.getTaskState().toString() + " Prio: " + bt.getPriority())
+        }
+    }
```

Comment:
I'd rather this be sent as one message with `\n` in order to prevent another chat message going in between